### ISSUE TITLE
Missing support for non 2D texture types in ofTexture

### ISSUE THREAD

**Issue by m1keall1son** [openframeworks/openFrameworks#6143]
Unless i'm mistaken, I can't find any support for non-2D texture types. 

missing:
GL_TEXTURE_2D_ARRAY,
GL_TEXTURE_3D,
GL_TEXTURE_1D,
GL_TEXTURE_1D_ARRAY,
GL_TEXTURE_CUBE_MAP,
GL_TEXTURE_CUBE_MAP_ARRAY,

It would make most sense to separate out the common functionality into an `ofTextureBase` base class and derive into something like `ofTexture1D`, `ofTexture2D`, `ofTexture3D` and `ofTextureCubeMap`.  For backwards compatibility alias `ofTexture2D` to `ofTexture`.

**Comment by roymacdonald** [comment]
Not really. There is like the scafolding for supporting it but not implemented yet.
I've used in the past [this](https://github.com/timscaffidi/ofxVolumetrics/blob/master/src/ofxTexture3d.h) to successfully access 3d textures. I think that it might need some refactoring so it works with the current OF implementation

**Comment by m1keall1son** [comment]
Thanks for linking that!  that will def help in the short term.  hopefully
I'll have a chance to take a pass at integrating the other texture types
in.

**Comment by roymacdonald** [comment]
So [here](https://github.com/roymacdonald/ofxVolumetrics.git) is what i've used in the past. I think that the last time I used it was like a year ago. Hope this helps

**Comment by arturoc** [comment]
this branch in this fork is more up to date and should work with 0.10 right away or with minimal changes https://github.com/Entropy/ofxVolumetrics/tree/feature-refactor

1d textures are supported if you use allocate with:

```
void allocate(const ofTextureData & textureData);
```

and specify the target in the texture data struct then just load data with a 1 line ofPixels.

For cubemaps there's:

https://github.com/andreasmuller/ofxCubeMap

which hasn't been updated in a while but should work with minimal changes since the basics for creating and loading data in textures haven't changed in a while

But yes would be good to have core support for more texture types

**Comment by arturoc** [comment]
oh sorry @roymacdonald that you posted your own repo for ofxVolumetrics which is more recently updated

**Comment by roymacdonald** [comment]
@arturoc no problem. Im not sure which one is updated most recently though.
